diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -517,11 +531,37 @@ data class ReplacementResult(val oldNodes: List<ASTNode>, val newNodes: List<AST
  * checks that this one node is placed after the other node in code (by comparing lines of code where nodes start)
  */
 fun ASTNode.isGoingAfter(otherNode: ASTNode): Boolean {
-    val thisLineNumber = this.lineNumber()
-    val otherLineNumber = otherNode.lineNumber()
+    val thisLineNumber = this.calculateLineNumber()
+    val otherLineNumber = otherNode.calculateLineNumber()
 
     require(thisLineNumber != null) { "Node ${this.text} should have a line number" }
     require(otherLineNumber != null) { "Node ${otherNode.text} should have a line number" }
 
     return (thisLineNumber > otherLineNumber)
 }
+
+fun ASTNode.getLineNumber(isFixMode: Boolean): Int? =
+    if (!isFixMode) {
+        lineNumber()
+    } else {
+        calculateLineNumber()
+    }
+
+/**
+ * This function calculates line number instead of using cached values.
```

Comment:
I thought we discussed it and decided that ktlint calculates everything correctly.
So why it is needed?

Reply:
We discussed it regarding warning messages, but we also use line number in some rules (e.g., method `ASTNode.isBefore`), and when AST has already been mutated by previous rules, it leads to `ArrayIndexOutOfBoundsException` (see #371).